src/core/a-entity.js

```diff
@@ -310,6 +310,11 @@ class AEntity extends ANode {
     // Initialize dependencies first
     this.initComponentDependencies(componentName);
 
+    // If component is sceneOnly check the entity is the scene element
```

Comment:
Wonder if it would be better to add this logic in the component constructor. Also the multiplicity check below. to keep things tidier here

Reply:
The component does have all the information needed to perform these checks, so I moved the checks. Inside `component.js` is now also the only place the `sceneOnly` and `multiple` flag are read, which is nice.